File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/22.java
def afd_diferenca_multiplo_3(s):
    diferenca = 0
    for char in s:
        if char == 'a':
            diferenca += 1
        elif char == 'b':
            diferenca -= 1
    return diferenca % 3 == 0

# Testes
print(afd_diferenca_multiplo_3("aaabbb"))  # True
print(afd_diferenca_multiplo_3("aab"))    # False

  }
  }

File: Computabilidade-e-Complexidade-de-Algoritmos-main/golang/22.go
package main
import "fmt"

func diferencaMultiplo3(s string) bool {
    diferenca := 0
    for _, char := range s {
        if char == 'a' {
            diferenca++
        } else if char == 'b' {
            diferenca--
        }
    }
    return diferenca%3 == 0
}

func main() {
    fmt.Println(diferencaMultiplo3("aaabbb"))  // True
    fmt.Println(diferencaMultiplo3("aab"))     // False
}
